### PR TITLE
Build & test cAdvisor with go 1.7.1

### PR DESCRIPTION
**Pull request by timstclair**
Fixes https://github.com/google/cadvisor/issues/1432

/cc @dashpole

### CODE REVIEW

**Comment by timstclair** [comment]
Oops, I guess 1.7.2 isn't out yet. Let's go with 1.7.1.

File: build/jenkins_e2e.sh [hunk 1]
```diff
@@ -31,8 +31,11 @@ if [[ ! -z "$(git diff --name-only pages)" ]]; then
   exit 1
 fi
 
-make
-go build -tags test github.com/google/cadvisor/integration/runner
+# Build & test with go 1.7
+docker run --rm \
```
Comment (vishh) [inline build/jenkins_e2e.sh]: Wouldn't it better to make the local and jenkins build be the same?

Comment (timstclair) [inline build/jenkins_e2e.sh]: I'm not sure how to do that without cloning every make rule. Is that worth it? Locally I can control my go version, but with jenkins this is the recommended way.

**Comment by timstclair** [comment]
Now that Kubernetes is requiring go1.7, can we merge this as is? We can update the local builds in a follow up.

File: build/jenkins_e2e.sh (hunk 1 above)
Comment (vishh) [inline build/jenkins_e2e.sh]: One option is to detect if we are in a containerized environment by setting an environment variable. If we are inside a container, then run the make command directly. If not, then invoke make run_container with the actual make command as an argument.

**Comment by vishh** [comment]
@timstclair SGTM

**Pull request merged**